### PR TITLE
#237: Support extra MDU sections

### PR DESCRIPTION
**Pull request by priscavdsluis**

### CODE REVIEW

File: hydrolib/core/io/mdu/models.py [hunk 1]
```diff
@@ -502,12 +503,169 @@ def is_intermediate_link(self) -> bool:
...
+    The `[Calibration]` section in an MDU file.
+
+    This model is typically referenced under [FMModel][hydrolib.core.io.mdu.models.FMModel]`.calibration`.
+
+    All lowercased attributes match with the [Calibration] input as described in
+    [UM Sec.A.3](https://content.oss.deltares.nl/delft3d/manuals/D-Flow_FM_User_Manual_1D2D.pdf#section.A.3).
+    """
+
+    _header: Literal["Calibration"] = "Calibration"
+    usecalibration: bool = Field(0, alias="UseCalibration")
```
Comment (arthurvd) [inline hydrolib/core/io/mdu/models.py:517]: 0->False?

File: hydrolib/core/io/mdu/models.py [hunk 2]
```diff
@@ -502,12 +503,169 @@ def is_intermediate_link(self) -> bool:
...
+    definitionfile: Optional[Path] = Field(None, alias="DefinitionFile")
```
Comment (arthurvd) [inline hydrolib/core/io/mdu/models.py:518]: We have to remember to change `Optional[Path]` -> `DiskOnlyFileModel` after #281 is closed?

File: hydrolib/core/io/mdu/models.py [hunk 3]
```diff
@@ -502,12 +503,169 @@ def is_intermediate_link(self) -> bool:
...
+class Calibration(INIBasedModel):
+    """
+    The `[Calibration]` section in an MDU file.
+
+    This model is typically referenced under [FMModel][hydrolib.core.io.mdu.models.FMModel]`.calibration`.
+
+    All lowercased attributes match with the [Calibration] input as described in
+    [UM Sec.A.3](https://content.oss.deltares.nl/delft3d/manuals/D-Flow_FM_User_Manual_1D2D.pdf#section.A.3).
+    """
+
```
Comment (arthurvd) [inline hydrolib/core/io/mdu/models.py:515]: Nowhere we have any comments in this FMModel components. Maybe let's do that in a separate enhancement issue.

File: hydrolib/core/io/mdu/models.py [hunk 4]
```diff
@@ -502,12 +503,169 @@ def is_intermediate_link(self) -> bool:
...
+    _header: Literal["Calibration"] = "Calibration"
+    usecalibration: bool = Field(0, alias="UseCalibration")
+    definitionfile: Optional[Path] = Field(None, alias="DefinitionFile")
+    areafile: Optional[Path] = Field(None, alias="AreaFile")
+
+
+class InfiltrationMethod(IntEnum):
+    """
+    Enum class containing the valid values for the Infiltrationmodel
+    attribute in the `Grw` class.
```
Comment (arthurvd) [inline hydrolib/core/io/mdu/models.py:525]: `GRrw` -> [Groundwater][hydrolib.core.io.mdu.models.Groundwater]

File: hydrolib/core/io/mdu/models.py [hunk 5]
```diff
@@ -502,12 +503,169 @@ def is_intermediate_link(self) -> bool:
...
+    h_aquiferuni: Optional[float] = Field(20.0, alias="h_aquiferuni")
+    bgrwuni: Optional[float] = Field(None, alias="bgrwuni")
+    h_unsatini: Optional[float] = Field(0.2, alias="h_unsatini")
+    sgrwini: Optional[float] = Field(None, alias="sgrwini")
+
+
+class ProcessFluxIntegration(IntEnum):
+    """
+    Enum class containing the valid values for the ProcessFluxIntegration
+    attribute in the `Processes` class.
```
Comment (arthurvd) [inline hydrolib/core/io/mdu/models.py:565]: `Processes` -> [Processes][hydrolib.core.io.mdu.models.Processes]

File: hydrolib/core/io/mdu/models.py [hunk 6]
```diff
@@ -502,12 +503,169 @@ def is_intermediate_link(self) -> bool:
...
+    dtmassbalance: Optional[float] = Field(0.0, alias="DtMassBalance")
+    processfluxintegration: Optional[float] = Field(
+        ProcessFluxIntegration.WAQ, alias="ProcessFluxIntegration"
+    )
+    wriwaqbot3doutput: Optional[bool] = Field(False, alias="Wriwaqbot3Doutput")
+    volumedrythreshold: Optional[float] = Field(1e-3, alias="VolumeDryThreshold")
+    depthdrythreshold: Optional[float] = Field(1e-3, alias="DepthDryThreshold")
+
+
+class ThreeDType(IntEnum):
```
Comment (arthurvd) [inline hydrolib/core/io/mdu/models.py:600]: Consider renaming ThreeDType to Particles3DType (it is an enum specific for Particles)

File: hydrolib/core/io/mdu/models.py [hunk 7]
```diff
@@ -502,12 +503,169 @@ def is_intermediate_link(self) -> bool:
...
+    addtracer: Optional[bool] = Field(False, alias="AddTracer")
+    starttime: Optional[float] = Field(0.0, alias="StartTime")
+    timestep: Optional[float] = Field(0.0, alias="TimeStep")
+    threedtype: Optional[ThreeDType] = Field(ThreeDType.DepthAveraged, alias="3Dtype")
+
+
+class VegetationModelNr(IntEnum):
+    """
+    Enum class containing the valid values for the VegetationModelNr
+    attribute in the `Vegetation` class.
```
Comment (arthurvd) [inline hydrolib/core/io/mdu/models.py:633]: `Vegetation` -> [Vegetation][hydrolib.core.io.mdu.models.Vegetation]

File: hydrolib/core/io/mdu/models.py [hunk 8]
```diff
@@ -502,12 +503,169 @@ def is_intermediate_link(self) -> bool:
...
+    """
+
+    _header: Literal["Veg"] = "Veg"
+
+    vegetationmodelnr: Optional[VegetationModelNr] = Field(
+        VegetationModelNr.No, alias="Vegetationmodelnr"
+    )
+    clveg: Optional[float] = Field(0.8, alias="Clveg")
+    cdveg: Optional[float] = Field(0.7, alias="Cdveg")
+    cbveg: Optional[float] = Field(0.7, alias="Cbveg")
```
Comment (arthurvd) [inline hydrolib/core/io/mdu/models.py:657]: default 0.0

**Review by arthurvd — CHANGES_REQUESTED** [review]
Hi @priscavdsluis , review almost ok, minor changes needed. I already fixed the hyperlinks to UM appendices.

File: hydrolib/core/io/mdu/models.py (hunk 2 above)
Comment (priscavdsluis) [reply hydrolib/core/io/mdu/models.py:518]: Yes! :)

File: hydrolib/core/io/mdu/models.py (hunk 3 above)
Comment (priscavdsluis) [reply hydrolib/core/io/mdu/models.py:515]: Issue: #288

**Review by arthurvd — COMMENTED** [review]
Hi @priscavdsluis, all changes now OK. Maybe a good idea to add these added sections to one of our test input model files, to add some unit testing?

**Review by arthurvd — APPROVED** [review]

**Pull request merged**